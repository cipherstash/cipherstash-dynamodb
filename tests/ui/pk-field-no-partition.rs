use cipherstash_dynamodb::Encryptable;

#[derive(Debug, Encryptable)]
struct User {
    pk: String,
    #[partition_key]
    #[cipherstash(query = "exact", compound = "email#name")]
    email: String,
    #[cipherstash(query = "prefix", compound = "email#name")]
    name: String,
}

fn main() {}
