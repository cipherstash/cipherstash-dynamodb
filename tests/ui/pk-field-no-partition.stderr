error: field named 'pk' must be annotated with #[partition_key]
 --> tests/./ui/pk-field-no-partition.rs:5:5
  |
5 |     pk: String,
  |     ^^^^^^^^^^
